modin/pandas/test/test_series.py

```diff
@@ -3475,6 +3475,13 @@ def test_to_numpy(data):
     assert_array_equal(modin_series.to_numpy(), pandas_series.to_numpy())
 
 
+def test_to_numpy_dtype():
+    modin_series, pandas_series = create_test_series(test_data["float_nan_data"])
+    assert_array_equal(
+        modin_series.to_numpy(dtype="int64"), pandas_series.to_numpy(dtype="int64")
```

Comment:
```suggestion
        modin_series.to_numpy(dtype="int64"),
        pandas_series.to_numpy(dtype="int64"),
        strict=True,
```
you have to pass [`strict=True`](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_array_equal.html) here, as otherwise it won't raise on different dtypes:
```python
>>> assert_array_equal(np.array([1]), np.array([1.0]))
>>> # no errors
>>> assert_array_equal(np.array([1]), np.array([1.0]), strict=True)
AssertionError:
Arrays are not equal

(dtypes int32, float64 mismatch)
 x: array([1])
 y: array([1.])
```
```

Reply:
Yes, but this test still did not pass without new changes, because there was no actual type conversion and the values in the internal buffers of the numpy arrays did not match. So do I still need to add `strict=True` or not?

Reply:
I think you better add `strict=True` anyway